### DISCUSSION TITLE
Pawn is not moving and not sure what's wrong

### DISCUSSION THREAD

**Discussion by CharlesHorsman-TwistedDev** [midgen/uesvon#33] [Q&A]
Hey guys! New to UE5. I'm trying to make a fish move around. I've followed the BP steps listed but I'm not sure what's wrong. I've been struggling with this the last 4 days and I've tried a bunch of random crap with no luck. As you can see below I have the SVON navigation added to my AI. I also have the SVONVolume setup with my pawn inside of it. I did click generate
![image](https://github.com/midgen/uesvon/assets/59493530/b26301b7-53a3-441a-af56-f764494b88dc)
I tell it to run my Behavior tree
![image](https://github.com/midgen/uesvon/assets/59493530/dd12b1be-51eb-46b4-b941-40943bf6bd77)
From here I have a very simple tree
![image](https://github.com/midgen/uesvon/assets/59493530/80a507cd-881b-46c9-9e26-5438deb8f226)
But nothing happens. The Fish doesn't move

the only thing I can think of is that I haven't actually given the fish a location to move to. I've played with the SVON_MoveTo quite a bit but nothing ever seems to happen. Is there any chance a very simple project or youtube video that uses this plugin that I can use to try and figure out why my basic fish isn't moving?

**Comment by midgen** [comment]
There's some example blueprints and a demo in the Cashdemo project :

https://github.com/midgen/CashDemo

The problem you have there is that your pawn is trying to move to itself.

The demo has this blueprint in the pawn : 

![image](https://github.com/midgen/uesvon/assets/7430875/a937196f-73b7-48d8-a61e-3aa93d6a319c)

This just gets the first player and sets that as the TargetActor in the Blackboard.

It then runs this Behaviour Tree

![image](https://github.com/midgen/uesvon/assets/7430875/637c1def-bf54-4bc7-b114-8b590b340cbb)

The decorator just says only run this if the target is greater than X distance away. 
The service has this BP code :

![image](https://github.com/midgen/uesvon/assets/7430875/4ae253d0-cb61-4d88-8b69-2dce627a2ce2)

It is taking the TargetActor from the blackboard and setting the TargetPosition to be the TargetActor's transform. 

The SVONMoveTo node then just moves to the TargetPosition.

It's very crude but sufficient for a demo!

**Reply by CharlesHorsman-TwistedDev** [reply]
Thank you so much for the reply! I'll take a look at the demo today and see what all i did wrong. 

Might I suggest adding the demo as a link in your wiki? Your wiki for the steps weren't super clear and i couldn't find anything related to this plugin on google to help me solve what I was doing wrong.

Thanks again for the quick answer though!
